openpifpaf/network/losses/composite.py

```diff
@@ -82,15 +87,20 @@ def _confidence_loss(self, x_confidence, t_confidence):
         ce_loss = self.confidence_loss(x_confidence, bce_target)
         if self.prescale != 1.0:
             ce_loss = ce_loss * self.prescale
+        weights_torch = torch.tensor(self.weights, device=bce_masks.device).unsqueeze(0).unsqueeze(2).unsqueeze(3)  # TODO find a nicer way to assign the weights to the correct GPU
```

Comment:
This tensor is the same in every forward pass. Please move it to the constructor and use it in place of `self.weights`. Once that is done, you don't need to assign the `device` anymore. I would first create a tensor initialized to ones (`torch.ones`) with shape `[1, n_fields, 1, 1]`. Then if `w` is not None, you set the entries with something like `self.weights[0, :, 0, 0] = w`.